src/login/AuthProviderBase.ts

```diff
@@ -48,7 +49,14 @@ export abstract class AuthProviderBase<TLoginResult> {
 		}
 
 		const nonce: string = randomBytes(16).toString('base64');
-		const { server, redirectPromise, codePromise } = createServer(nonce);
+		const { server, redirectPromise, codePromise, codeTimer } = createServer(context, nonce);
+
+		cancellationToken.onCancellationRequested(() => {
+			server.close(error => error && ext.outputChannel.appendLog(parseError(error).message));
+			clearTimeout(codeTimer);
+			context.telemetry.properties.serverClosed = 'true';
+			ext.outputChannel.appendLog(localize('azure-account.serverClosed', 'Server closed.'));
```

Comment:
I don't think the average user will know what "server closed" means, and it might lead to confusion (e.g. "Is Microsoft's server closed?"). Might be better to just say that the auth process has been cancelled.